Document exported API types (#452)
grammyjs/website: site/docs/guide/api.md

This is easy to forget, hard to debug, and it breaks type inference.
grammY allows you to specify objects consistently across the API, and makes sure that the right properties are serialized on the fly before sending them.

### Type Definitions for the API

grammY ships with complete type coverage of the Bot API.
The [`@grammyjs/types`](https://github.com/grammyjs/types) repository contains the type definitions that grammY uses internally.
These type definitions are also exported so you can use them in your own code.

#### Type Definitions on Node.js

On Node.js, you need to import the types from `grammy/types`.
For example, you get access to the `Chat` type like this:

```ts
import { type Chat } from "grammy/types";
```

Officially, Node.js only supports importing from sub-paths properly since Node.js 16.
Consequently, TypeScript requires the `moduleResolution` to be set to `node16` or `nodenext`.
Adjust your `tsconfig.json` accordingly and add the highlighted line:

> ```jsonc{4}

Reviewer: ` ```json{4}` instead of ` ```jsonc{4}`?

{
  "compilerOptions": {
    // ...
    "moduleResolution": "node16"
    // ...
  }
}
```

However, this can sometimes also work without adjusting your TypeScript configuration.

::: warning Wrong Autocomplete

If you do not change your `tsconfig.json` file as described above, it may happen that your code editor suggests in autocomplete to import types from `grammy/out/client` or something.
**All paths starting with `grammy/out` are internal. Do not use them.**
They could be changed arbitrarily at any point in time, so we strongly advise you to import from `grammy/types` instead.

:::

#### Type Definitions on Deno

> On Deno, you can simply import type definitions from `types.ts` instead of `mod.ts`.

Reviewer 2: ```suggestion
On Deno, you can simply import type definitions from `types.ts`:
```

Author: I agree the phrasing is bad. Can we try to keep the point that the file is located next to `mod.ts`?
```suggestion
On Deno, you can simply import type definitions from `types.ts`, which is right next to `mod.ts`.
```

Reviewer 2: You can do that, but remove the comma. and make the dot a semi-colon.

Author: ```suggestion
On Deno, you can simply import type definitions from `types.ts` which is right next to `mod.ts`:
```
?

Reviewer 2: Exactly
